### PR TITLE
Backport github: Enable github actions

### LINKED ISSUE #293: backport: enable github actions for stable 1.12

### PR DESCRIPTION
**Pull request by GabyCT**
This PR enables github actions for the proxy repository.

Fixes #293

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+on: ["pull_request"]
+name: Static checks
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.13.x, 1.14.x, 1.15.x]
+        os: [ubuntu-18.04]
```
Comment (jodh-intel) [inline .github/workflows/main.yml:8]: Can we use 20.04 out of interest?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        echo "TRAVIS_PULL_REQUEST_BRANCH: ${TRAVIS_PULL_REQUEST_BRANCH}"
+        echo "TRAVIS_PULL_REQUEST_SHA: ${TRAVIS_PULL_REQUEST_SHA}"
+        echo "TRAVIS: ${TRAVIS}"
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Setup travis references
+      run: |
+        echo "TRAVIS_BRANCH=${TRAVIS_BRANCH:-$(echo $GITHUB_REF | awk 'BEGIN { FS = \"/\" } ; { print $3 }')}"
```
Comment (jodh-intel) [inline .github/workflows/main.yml:36]: This block only echo's a string - is that what we want? Also, the previous `run` block uses a different format (no equals sign).

File: .github/workflows/main.yml (hunk 1 above)
Comment (GabyCT) [reply .github/workflows/main.yml]: @jodh-intel changes applied

**Review by devimc — APPROVED** [review]
thanks @GabyCT

**Pull request merged**